src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/TypeInfoResolverFunctionalTests.cs

```diff
@@ -697,24 +696,25 @@ public override JsonTypeInfo GetTypeInfo(Type type, JsonSerializerOptions option
                 if (jsonTypeInfo.Kind == JsonTypeInfoKind.Object &&
                     type.GetCustomAttribute<DataContractAttribute>() is not null)
                 {
-                    jsonTypeInfo.Properties.Clear(); // TODO should not require clearing
-
-                    IEnumerable<(PropertyInfo propInfo, DataMemberAttribute attr)> properties = type
-                        .GetProperties(BindingFlags.Instance | BindingFlags.Public)
-                        .Where(propInfo => propInfo.GetCustomAttribute<IgnoreDataMemberAttribute>() is null)
-                        .Select(propInfo => (propInfo, attr: propInfo.GetCustomAttribute<DataMemberAttribute>()))
-                        .OrderBy(entry => entry.attr?.Order ?? 0);
+                    jsonTypeInfo.Properties.Clear();
```

Comment:
since you're effectively doing almost everything from scratch it probably makes sense to also do JsonTypeInfo from scratch and implement it as IJsonTypeInfoResolver which returns null for types it can't handle and on the usage site combine that with DefaultJsonTypeInfoResolver

Reply:
file an issue with any remaining work

Reply:
I get that extending this pattern to other functional tests would be nice to have, but honestly don't see why we need to do this now? Do you have concerns about test coverage? Would you like to maybe contribute another PR that adds additional function tests? We have a checkmark for that in #71797.